src/classes/services/BoutiqueResourceService.cls

```diff
@@ -5,9 +5,15 @@
 public with sharing class BoutiqueResourceService {
 
     public static Boolean isPhoneNumberValid(String phoneNumber) {
-        if (phoneNumber.length() != 10 || !phoneNumber.isNumeric() || String.isBlank(phoneNumber)) {
+
+        Pattern phonePattern = Pattern.compile('[0-9]{10}');
+
+        Matcher phoneMatcher = phonePattern.matcher(phoneNumber);
+
+        if (!phoneMatcher.matches()) {
             return false;
         }
+
         return true;
```

Comment:
MINOR
extract regex to constant
simplify to `return Pattern.matches(PHONE_PATTERN, workPhone);`

are you sure we do not need ` String.isNotBlank(workPhone)` anymore?